packages/components/src/internal/components/editable/models.ts

```diff
@@ -704,24 +702,21 @@ export class EditorModel
                         return m;
                     }
 
-                    // data maps have keys that are display names/captions. We need to convert to the
-                    // encoded keys used in our filters to match up with values from the forms.
-                    const key_ = encode ? encodePart(key) : key;
-                    return m.set(key_, editorData);
-                }, Map<any, any>());
+                    return m.set(key, editorData);
+                }, Map<string, any>());
 
-                if (!updates) {
+                if (!queryInfo || !updates) {
                     return returnMap;
                 }
 
-                if (!idsNotToUpdate || idsNotToUpdate.indexOf(parseInt(id, 10)) < 0 || !fieldsNotToUpdate) {
-                    return returnMap.merge(updates);
-                }
-
-                let trimmedUpdates = Map<any, any>();
+                let trimmedUpdates = Map<string, any>();
+                const isNotUpdateId = idsNotToUpdate && idsNotToUpdate.indexOf(parseInt(id, 10)) > -1;
                 updates.forEach((value, fieldKey) => {
-                    if (fieldsNotToUpdate.indexOf(fieldKey.toLowerCase()) < 0) {
-                        trimmedUpdates = trimmedUpdates.set(fieldKey, value);
+                    const col = queryInfo.getColumn(fieldKey);
+                    const isFieldNotToUpdate =
+                        fieldsNotToUpdate && fieldsNotToUpdate.indexOf(fieldKey.toLowerCase()) > -1;
```

Comment:
There isn't anything that is going to enforce that the fields put in `fieldsNotToUpdate` are lower cased already. Consider defensively copying the array into a lower-cased collection so it is guaranteed things are lowered.